### ISSUE TITLE
Disabled on new pages

### ISSUE THREAD

**Issue by jcancellier** [birchill/10ten-ja-reader#1172]
On iOS safari, I have to go to the extensions menu and enable the extension every time I go to a new page. Is this a bug or intended? If intended I think it would be worth it to add a feature where it’s always visible and enabled by default.

**Comment by birtles** [comment]
Hi @jcancellier!

Yes, it's intended but I agree it's not great.

When Safari introduced Web Extensions they encouraged authors to focus on making their extensions more strict on privacy so we made 10ten use "Active tab" mode. That means 10ten can't see the Web page you're browsing unless you explicitly turn it on for that page but once you turn it on, it should be available for all pages on the same _site_ for that session.

We should really fix it so that on all platforms it starts in "Active tab" mode but then you can enable it permanently for certain sites or all sites. I'm just afraid of breaking things so I haven't tried doing that yet. Thanks for filing this issue, though, it helps me to prioritize this work.

**Comment by gamrah** [comment]
Just wanted to +1 this request,
Love the extension, but there's definitely a lot of re-opening required at the moment. You mentioned "once you turn it on, it should be available for all pages on the same site for that session" but that doesn't appear to be the case in my experience. Sub-domains seem to break that logic e.g. news.yahoo.co.jp and say approach.yahoo.co.jp so let's say the aggregator pulls you to a different website and you back out to the homepage again, that's another reload.

I'm not sure where this issue would fall on your priority list, but I think iOS users would really appreciate if you could get to this at some point! Happy to further test if you think it'll help

**Comment by birtles** [comment]
Thank you! The +1s are very helpful. I'm sorry about the state of Safari. I need to fix it, but it looks very hard and very risky so I've been deferring it until after other more urgent work. Upvoting this issue however really does help me to prioritize this work, however. Feel free to keep reminding me if nothing seems to be happening here.

**Comment by esgibtshoosh** [comment]
Hey I have the same issue on Safari. Is this still being considered?

**Comment by birtles** [comment]
> Hey I have the same issue on Safari. Is this still being considered?

Yes, it's still on the radar but I've no progress to report yet, unfortunately.

**Comment by AlexW00** [comment]
this "feature" makes 10ten on safari **unusable**.

>  it should be available for all pages on the same site for that **_session_**.

while it could be argued that "once enabled for a page, it will stay active" could be a reasonable feature, the status quo is "once enabled for a page, it will stay active **until you close the tab**" which is insane tbh.

the status quo requires users to click on the 10ten icon **every time they open a new tab**, which happens **all the time**

for me, this friction is just too much. I am honestly surprised, that this "feature" isn't a high prio bug instead...
